src/main/java/edu/harvard/iq/dataverse/MailServiceBean.java

```diff
@@ -496,6 +496,19 @@ public String getMessageTextBasedOnNotification(UserNotification userNotificatio
                     version.getDataset().getOwner().getDisplayName(),  getDataverseLink(version.getDataset().getOwner()), optionalReturnReason};
                 messageText += MessageFormat.format(pattern, paramArrayReturnedDataset);
                 return messageText;
+                
+            case WORKFLOW_SUCCESS:
+                version =  (DatasetVersion) targetObject;
+                pattern = BundleUtil.getStringFromBundle("notification.email.workflow.success");
+                String[] paramArrayWorkflowSuccess = {version.getDataset().getDisplayName(), getDatasetLink(version.getDataset()), comment};
+                messageText += MessageFormat.format(pattern, paramArrayWorkflowSuccess);
+                return messageText;
+            case WORKFLOW_FAILURE:
+                version =  (DatasetVersion) targetObject;
+                pattern = BundleUtil.getStringFromBundle("notification.email.workflow.failure");
+                String[] paramArrayWorkflowFailure = {version.getDataset().getDisplayName(), getDatasetLink(version.getDataset())};
```

Comment:
Am I reading this correctly, that it's not populating the 3rd parameter of `notification.email.workflow.failure` pattern here? 
So, should probably be 
`{version.getDataset().getDisplayName(), getDatasetLink(version.getDataset()), comment};`
(it's probably more important to show the "reason" comment in the failure email, than in the success one...)

Reply:
Good catch - not sure why I left it out - perhaps I just tested failure prior to adding the messaging part and didn't update it later. In any case, added now.